Created Dropdown to select with search option

prahe184 opened the pull request:
##Description:

I have created select.tsx and selectstories.tsx files and wrote the dropdown with select option and search option and added some code in page.tsx to render dropdown 

Closes #28

babarkamal114 reviewed (commented):
Great work and congrats on your first PR.

Here are some suggestions to improve the component's readability and make it production-ready:

**1. Missing Variant System Support**
We use `class-variance-authority` for styling.
Create a new file under the variants/ directory named select-variants.ts.
Define all styling variants there.
Refer to other components’ variant files for structure.

`2. Missing WCAG Compliance`
The component currently lacks key accessibility features:
No aria-label or aria-labelledby for screen readers
Missing keyboard support (e.g., arrow key navigation)
Focus ring is not clearly visible


**3. No Controlled Mode**
Right now, the value is managed internally.
Add support for controlled usage via value and onChange props.

**4. No Custom Option Rendering**
The component only renders plain text options.
Allow rendering custom JSX, like icons, tags, or avatars, for better flexibility.

babarkamal114 commented on src/app/page.tsx:
```diff
@@ -16,6 +17,18 @@ export default function Home() {
           and keep your team aligned effortlessly.
         </Typography>
+        <Select
+          label="Select your option"
```
Remove the Select from page.tsx since we're using Storybook. If you want to test it, use Storybook. I am creating a temporary landing page here.

prahe184 replied:
Done Sir

prahe184 commented:
I will fix them and add commit Sir

babarkamal114 commented on src/components/ui/select.tsx:
```diff
@@ -0,0 +1,132 @@
+  value: string;
+};
+
+type SelectProps = {
```
Move this to types/components.ts.
Replace sizes and color with the following props:

label?: string  
options: Option[]  
placeholder?: string  
icon?: LucideIcon  
error?: string  
loading?: boolean  
size?: Size  
color?: Color  
multiple?: boolean  
onChange: (selected: Option | Option[] | null) => void

prahe184 replied:
I am seeing so many red squiggles after changing it to component.ts Sir. I'm stuck and not able to find How can I figure it out

babarkamal114 replied:
That's alright. Whatever changes you've made, push them. I will fix it in your PR.

prahe184 replied:
Done Sir

babarkamal114 reviewed (changes requested):
Here are a few comments. You can resolve them first, and I’ll help you refactor the component afterward.

babarkamal114 commented:
Hey @prahe184 ,
I’ve renamed the branch and switched to another PR.
I also made some changes to the one you worked on.

Could you please review them, go through the code, and leave comments if you spot anything that needs improvement or fixing?

https://github.com/babarkamal114/cobhub/pull/44

Thanks!

prahe184 commented:
> Hey @prahe184 , I’ve renamed the branch and switched to another PR. I also made some changes to the one you worked on.
> 
> Could you please review them, go through the code, and leave comments if you spot anything that needs improvement or fixing?
> 
> #44
> 
> Thanks!

Hey @babarkamal114 , I've gone through the code Sir. No need to make any changes Sir

prahe184 reviewed (commented):
Everything is working Fine Sir
